src/streaming/controllers/ScheduleController.js

```diff
@@ -224,8 +224,10 @@ function ScheduleController(config) {
         if (!type || !currentRepresentation) {
             return true;
         }
+        const segmentDuration = currentRepresentation && currentRepresentation.segmentDuration && !isNaN(currentRepresentation.segmentDuration) ? currentRepresentation.segmentDuration : 0;
         const bufferLevel = dashMetrics.getCurrentBufferLevel(type);
-        return bufferLevel < getBufferTarget();
+
+        return bufferLevel + segmentDuration < getBufferTarget();
```

Comment:
Is there a risk that the set buffer target is less than the segment duration? If so, what should the behavior be? My thinking is that either (a) the target is implicitly set to segmentDuration, or that (b) _shouldBuffer should throw an exception. Or, do the checking everywhere the segmentDuration and buffer target are changed.

Reply:
Thanks @wschidol that is a good point. 

It does not make sense from an application point of view to set bufferTarget < segmentDuration but we should be robust here. I will add another PR for this. 

I think we should ignore the segment duration if it is larger than the buffer target. If we set segmentDuration = bufferTarget we risk stalling as well as the video playback will typically stall prior to the buffer being level being 0.

Reply:
As an addition, for CMAF low latency streaming the buffer target might in fact be smaller than the segment duration